Easydict/Swift/Service/LLMDerivs/SummaryService.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  SummaryService.swift
+//  Easydict
+//
+//  Created by Jerry on 2024-07-11.
+//  Copyright © 2024 izual. All rights reserved.
+//
+
+import Foundation
+import OpenAI
+
+@objc(EZSummaryService)
+class SummaryService: LLMDerivService {
+    public override func name() -> String {
+        NSLocalizedString("summary_service", comment: "")
+    }
+
+    public override func serviceType() -> ServiceType {
+        .summary
+    }
+
+    public override func translate(
+        _ text: String,
+        from: Language,
+        to: Language,
+        completion: @escaping (EZQueryResult, Error?) -> ()
+    ) {
+        let url = URL(string: endpoint)
+        let invalidURLError = EZError(type: .param, description: "`\(serviceType().rawValue)` endpoint is invalid")
+        guard let url, url.isValid else {
+            completion(result, invalidURLError)
+            return
+        }
+
+        var resultText = ""
+
+        result.isStreamFinished = false
+
+        // Set EZQueryTextType as translation since no formatting is needed
+        let queryType = EZQueryTextType.translation
+        let chatParam = LLMDerivParam(
+            text: text,
+            sourceLanguage: from,
+            serviceType: .summary
+        )
+
+        let chatHistory = serviceChatMessage(chatParam)
```

Comment:
"`SummaryService` and `PolishingService` have too much duplicate code; we can extract the prompt  code, add a new method called `chatHistory` to `BaseOpenAIService`, and they would only need to override this method.

Reply:
Sorry, I forgot there's already a method `serviceChatMessageModels()`, so `SummaryService` and `PolishingService` just need to override it.

Reply:
I've refactored the code, now we just need to simply rewrite the `chatMessageDicts()` method.

Reply:
Which part of `chatMessageDicts()` do we need to rewrite and what do we want to achieve?

Reply:
For custom AI tool service, you only need to rewrite `chatMessageDicts` to return a prompt in the format of `[[String: String]]`. 

In the `serviceChatMessageModels` method of `BaseOpenAIService`, we will automatically convert the dictionary format into the `ChatMessage` format required by `OpenAI`.

Please refer to the code for details.

<img width="1301" alt="image" src="https://github.com/user-attachments/assets/00820879-3ba9-4019-a700-247506388926">